### PR TITLE
SLING-10477 : Review usage of UserEntryHandler in RecollectorVaultPackageScanner

### PR DESCRIPTION
**Pull request by anchela**
@karlpauls , extended the user-group-serviceuser-convertion a bit to add assertions that ACE for users/groups are contained in the converted package (but not those for service users).

let's check if the usage of "UserEntryHandler" in RecollectorVaultPackageScanner is really needed.

### CODE REVIEW

**Review by karlpauls — APPROVED** [review]
@anchela, I added a test for packages of type CONTENT and removed the UserEntryHandler from the recollector. I think it is not needed there any more as it is nowadays listed in the services.

I removed the null check for the assembler as well as it now should not be needed anymore.

File: src/test/java/org/apache/sling/feature/cpconverter/ContentPackage2FeatureModelConverterTest.java [hunk 1]
```diff
@@ -1041,9 +1041,125 @@ public void testConvertPackageWithUsersGroupsAndServiceUsers() throws Exception
...
+                String policy = IOUtils.toString(in, StandardCharsets.UTF_8);
+                for (String principalName : expectedPrincipalNames) {
+                    assertTrue(policy.contains("rep:principalName=\""+principalName+"\""));
+                }
+                if (unExpectedPrincipalName != null) {
+                    assertFalse(policy.contains("rep:principalName=\""+unExpectedPrincipalName+"\""));
+                }
+            }
+        } 
+    }
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ContentPackage2FeatureModelConverterTest.java:1163]: now we have the same test assertions copied 3 times, right?
i will move that to a separate ticket and add the zip-file as a separate param.

File: src/main/java/org/apache/sling/feature/cpconverter/vltpkg/RecollectorVaultPackageScanner.java [hunk 2]
```diff
@@ -42,6 +43,7 @@ public RecollectorVaultPackageScanner(@NotNull ContentPackage2FeatureModelConver
         this.converter = converter;
         handlers = new EntryHandler[] {
                 new UsersEntryHandler(),
+                new GroupEntryHandler(),
```
Comment (anchela) [inline src/main/java/org/apache/sling/feature/cpconverter/vltpkg/RecollectorVaultPackageScanner.java:46]: that looks reasonable!

File: src/test/java/org/apache/sling/feature/cpconverter/ContentPackage2FeatureModelConverterTest.java (hunk 1 above)
Comment (karlpauls) [reply src/test/java/org/apache/sling/feature/cpconverter/ContentPackage2FeatureModelConverterTest.java:1163]: well, one of the 3 times has a slightly different expectation on the content (doesn't include /apps). Not sure that one is really needed anyways. The other two only differ in the demo zip, yes - makes sense as a param.

**Comment by anchela** [comment]
@karlpauls , i think we are good. merge?

**Pull request merged**